flight/mcu/lib/teensyShared/HighGSensor.cpp

```diff
@@ -1,12 +1,42 @@
 #include "HighGSensor.h"
 
-#include <ChRt.h>
+#include "dataLog.h"
+#include "pins.h"
 
-void HighGSensor::readReadings() {
-    auto data = KX->getAccelData();
+HighGSensor highG;
+
+void HighGSensor::update() {
+    chSysLock();
+    chMtxLock(&mutex);
+    auto data = KX.getAccelData();
     ax = data.xData;
     ay = data.yData;
     az = data.zData;
+    timestamp = chVTGetSystemTime();
+    dataLogger.pushHighGFifo((HighGData) {ax, ay, az, timestamp });
+
+    chMtxUnlock(&mutex);
+    chSysUnlock();
+}
+
+Acceleration HighGSensor::getAccel() {
+    return { ax, ay, az };
 }
 
-GForce HighGSensor::getGForce() { return {ax, ay, az}; }
+void HighGSensor::init() {
+    if (!KX.beginSPICore(KX134_CS, 1000000, SPI)) {
+        Serial.println("Failed to communicate with KX134. Stalling Program");
+        digitalWrite(LED_RED, HIGH);
+        while (true)
+            ;
+    }
```

Comment:
stalling the program shouldn't be a failure mode here, it should return an error code / boolean